front/src/modules/people/hooks/deleteItems.tsx

```diff
@@ -0,0 +1,70 @@
+import { useRecoilValue } from 'recoil';
+
+import { filtersScopedState } from '@/ui/filter-n-sort/states/filtersScopedState';
+import { Filter } from '@/ui/filter-n-sort/types/Filter';
+import { IconTrash } from '@/ui/icon/index';
+import { useSnackBar } from '@/ui/snack-bar/hooks/useSnackBar';
+import { useResetTableRowSelection } from '@/ui/table/hooks/useResetTableRowSelection';
+import { selectedRowIdsSelector } from '@/ui/table/states/selectedRowIdsSelector';
+import { TableContext } from '@/ui/table/states/TableContext';
+import { useRecoilScopedState } from '@/ui/utilities/recoil-scope/hooks/useRecoilScopedState';
+
+export function useDeleteItems({
+  handleDeleteItems,
+  id,
+}: {
+  handleDeleteItems: (val: any) => any;
+  id: string;
+}) {
+  const { enqueueSnackBar } = useSnackBar();
+  const selectedRowIds = useRecoilValue(selectedRowIdsSelector);
+
+  const resetRowSelection = useResetTableRowSelection();
+
+  const [_, setFilters] = useRecoilScopedState(
+    filtersScopedState,
+    TableContext,
+  );
+
+  function handleCancelClick() {
+    // Remove the filter that hides the rows that are being deleted
+    setFilters((prevFilters) =>
+      prevFilters.filter(({ type }) => type !== '_exclusion_projection'),
+    );
+  }
+
+  function handleTimeout() {
```

Comment:
We do the actual deletion when the Toast completes, or when it's closed(not cancelled) by the user